api/http/Server_test.go

```diff
@@ -157,13 +157,28 @@ func TestGrantedLicenseAffectsCountsAndDetails(t *testing.T) {
 		  }`), srv)
 
 	resp = runRequestWithServer(get("/v1alpha/orgs/aspian/licenses/smarts", "token"), srv)
-	assertJSONResponse(t, resp, 200, `{"seatsAvailable":19, "seatsTotal": 20}`)
+	assertJSONResponse(t, resp, 200, `{"seatsAvailable":1, "seatsTotal": 2}`)
 	resp = runRequestWithServer(get("/v1alpha/orgs/aspian/licenses/smarts/seats", "token"), srv)
 	assertJSONResponse(t, resp, 200, `{"users": [{"assigned":true,"displayName":"Okay User","id":"okay"}]}`)
 	resp = runRequestWithServer(get("/v1alpha/orgs/aspian/licenses/smarts/seats?filter=assignable", "token"), srv)
 	assertJSONResponse(t, resp, 200, `{"users":[{"assigned":false,"displayName":"Bad User","id":"bad"}]}`)
 }
 
+func TestOverAssigningLicensesFails(t *testing.T) {
```

Comment:
It feels a bit weird not having the "given" state defined within the scope of the test, i.e. that there are 2 seats and 0 of them have been assigned. This state is buried in mockAccessRepository() in runRequest()/srv instead.

It would be nice to have something like:
// given
srv = createTestServerWithSeats(2, 0)
// when
resp := runRequestWithServer(req, srv) 
// then
assert.Equal(t, 400, resp.StatusCode)

Reply:
+1, apart from this lgtm